src/software/thunderscope/gl/layers/gl_movement_field_test_layer.py

```diff
@@ -0,0 +1,90 @@
+import math
+from pyqtgraph.Qt.QtCore import Qt
+from software.thunderscope.gl.helpers.extended_gl_view_widget import MouseInSceneEvent
+from proto.import_all_protos import *
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+from software.logger.logger import create_logger
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+
+logger = create_logger(__name__)
+
+
+class GLMovementFieldTestLayer(GLLayer):
+    def __init__(
+        self, name: str, fullsystem_io: ProtoUnixIO, buffer_size: int = 5
+    ) -> None:
+        """Initialize the GLMovementLayer
+
+        :param name:            The displayed name of the layer
+        :param buffer_size:     The buffer size we have
+        :param fullsystem_io:   The fullsystem protounix io
+        """
+        super().__init__(name)
+
+        self.world_buffer: ThreadSafeBuffer = ThreadSafeBuffer(buffer_size, World)
+        self.fullsystem_io: ProtoUnixIO = fullsystem_io
+        self.friendly_robot_on_field = []
+
+    def mouse_in_scene_pressed(self, event: MouseInSceneEvent) -> None:
+        """Move to the point clicked
+
+        :param event: The event
+        """
+        if not self.visible():
+            return
+
+        if not event.mouse_event.modifiers() == Qt.KeyboardModifier.AltModifier:
+            return
+
+        point = event.point_in_scene
+        self.move_to_point(point)
+
+    def find_closest_robot(self, point):
+        # finding the closest robot on field
+        closest_robot = self.friendly_robot_on_field[0].id
+        for robot in self.friendly_robot_on_field:
+            logger.info(
+                "robot: {} point: {}".format(robot.current_state.global_position, point)
+            )
+
+    def move_to_point(self, point):
+        """Move to a point
+
+        :param point: the point we are commanding the robot to move to
+        """
+        if len(self.friendly_robot_on_field) == 0:
+            logger.warning("There are no friendly robots on the field.")
+            return
+
+        # finding the cloest to the f
+        self.find_closest_robot(point)
```

Comment:
you can pybind this function and use it instead: https://github.com/UBC-Thunderbots/Software/blob/b27a9b5ef24a7d5b48cee5f1a16471c94db2add2/src/software/world/team.h#L189